python/GangaLHCb/Lib/Applications/BenderBox.py

```diff
@@ -0,0 +1,520 @@
+#  Actually there are three applications under the single hat
+#  - BenderModule : classic Bender application: use a full power of bender 
+#  - BenderRun    : execute scripts in the context of ``bender'' environment
+#  - OstapRun     : execute scripts in the context of ``ostap'' environment
+#  @author Vladimir ROMANOVSKY Vladimir.Romanovskiy@cern.ch
+#  @author Vanya BELYAEV  Ivan.Belyaev@itep.ru       
+# =============================================================================
+"""Bender application for Ganga  (CMAKE-based version)
+Actually there are three applications under the single hat
+- BenderModule : classic Bender application: use a full power of bender 
+- BenderScript : execute scripts in the context of ``bender'' environment
+- Ostap : execute scripts in the context of ``ostap'' environment
+"""
+# =============================================================================
+import os
+from   Ganga.GPIDev.Lib.File.File      import ShareDir,File
+from   Ganga.GPIDev.Lib.File.LocalFile import LocalFile
+from   Ganga.GPIDev.Schema             import Schema, Version, SimpleItem, GangaFileItem, FileItem, ComponentItem
+from   Ganga.GPIDev.Base               import GangaObject
+
+# =============================================================================
+## @class Bender_module
+#  Helper class to define the main properties of Bender applicatiom
+#  - the name of Bender module to run
+#  - dictionary of parameters to be forwarder to <code>configure</code> method
+#  - number of event to process
+#
+#  Usage:
+#  @code 
+#  j.application = BenderBox ( tool = BenderModule ( module = 'the_path/the_module.py' , events = 1000 ) )
+#  @endcode 
+#  @author Vladimir ROMANOVSKY Vladimir.Romanovskiy@cern.ch
+#  @author Vanya BELYAEV  Ivan.Belyaev@itep.ru       
+class BenderModule(GangaObject):
+    """Helper class to define the main properties of Bender applicatiom
+    - the name of Bender module to run
+    - dictionary of parameters to be forwarder to <code>configure</code> method
+    - number of event to process
+
+    ======
+    Usage:
+    ======
+    
+    j.application = BenderBox ( tool = BenderModule ( module = 'the_path/the_module.py' , events = 1000 ) )
+
+    """    
+    _schema = Schema(Version(1, 0), {
+        'module'    : FileItem   (
+        defvalue  = File () ,
+        doc       = """The file with Bender module. It is expected that module contains methods ``configure'' & ``run'' with the proper signatures""") , 
+        'params'    : SimpleItem (
+        defvalue  = {} ,
+        typelist  = ['dict', 'str', 'int', 'bool', 'float'],
+        doc       = """The dictionary of parameters to be forwarded to ``configure'' method of the supplied Bender module""") , 
+        'events'    : SimpleItem ( defvalue=-1, typelist=['int'], doc= "Number of events to process"),
+        })
+    _category = 'BenderTool'
+    _name     = 'BenderModule'
+    
+    layout    = """#!/usr/bin/env python
```

Comment:
just a suggestion but I would be tempted to use the dedent function to allow this to be indented a bit.